controllers/operandrequest/reconcile_operator.go

```diff
@@ -185,11 +185,21 @@ func (r *Reconciler) createSubscription(ctx context.Context, cr *operatorv1alpha
 	return nil
 }
 
-func (r *Reconciler) updateSubscription(ctx context.Context, cr *operatorv1alpha1.OperandRequest, sub *olmv1alpha1.Subscription) error {
+func (r *Reconciler) updateSubscription(ctx context.Context, cr *operatorv1alpha1.OperandRequest, sub *olmv1alpha1.Subscription, key types.NamespacedName) error {
 
 	klog.V(2).Infof("Updating Subscription %s/%s ...", sub.Namespace, sub.Name)
 	cr.SetUpdatingCondition(sub.Name, operatorv1alpha1.ResourceTypeSub, corev1.ConditionTrue)
 
+	sub.Annotations[key.Namespace+"."+key.Name] = "true"
```

Comment:
@Daniel-Fan I believe we need to add a check to make sure if `sub.Annotations` exists to make it work for the upgrade situation.

Reply:
hmmm....I did not get the point.

If it is the upgrade scenario, after the ODLM is upgraded, it will reconcile all the resource instance, is that right? And it will add the `annotations` for each subscription based on `registry` in `OperandRequest`.

Is it the same as the fresh installation?

Reply:
Yes, but existing subscriptions don't have `Annotations` field. it will return Invalid memory address or nil pointer dereference issue.

We need to make a map first.

Reply:
Get it!
Tested it in a upgrade scenario. It indeed will cause the panic of nil pointer dereference.
Really thank you to point it out!